libs/dh/esett/shell/src/lib/outgoing-messages/filters/dh-filters.component.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * @license
+ * Copyright 2020 Energinet DataHub A/S
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License2");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output,
+} from '@angular/core';
+import { FormControl, FormGroup, ReactiveFormsModule } from '@angular/forms';
+import { TranslocoDirective } from '@ngneat/transloco';
+import { Subscription, debounceTime } from 'rxjs';
+
+import { WattFormChipDirective, WattFormFieldComponent } from '@energinet-datahub/watt/form-field';
+import { WattButtonComponent } from '@energinet-datahub/watt/button';
+import { WattDropdownComponent } from '@energinet-datahub/watt/dropdown';
+import { WattDateRangeChipComponent } from '@energinet-datahub/watt/datepicker';
+import { VaterSpacerComponent, VaterStackComponent } from '@energinet-datahub/watt/vater';
+
+import { DhOutgoingMessagesFilters } from '../dh-outgoing-messages-filters';
+
+// Map query variables type to object of form controls type
+type FormControls<T> = { [P in keyof T]: FormControl<T[P] | null> };
+type Filters = FormControls<DhOutgoingMessagesFilters>;
+
+/** Helper function for creating form control with `nonNullable` based on value. */
+const makeFormControl = <T>(value: T = null as T) =>
```

Comment:
Suggestion: Move this to a util lib, we have this in more than one place :-)

Reply:
Done :)